.azure-pipelines/third-party-test-suites.yml

```
@@ -0,0 +1,369 @@
+        sed -i "s|#wal_sender_timeout =|wal_sender_timeout = 3s #|" pgsql/PGDATA/postgresql.conf
+        sed -i "s|#synchronous_standby_names =|synchronous_standby_names = 'npgsql_test_sync_standby' #|" pgsql/PGDATA/postgresql.conf
+        sed -i "s|#synchronous_commit =|synchronous_commit = local #|" pgsql/PGDATA/postgresql.conf
+        pgsql/bin/pg_ctl -D pgsql/PGDATA -l logfile -o '-c max_prepared_transactions=10 -c ssl=true -c ssl_cert_file=../server.crt -c ssl_key_file=../server.key' start
+
+        # Configure test account
+        pgsql/bin/psql -U postgres -c "CREATE ROLE npgsql_tests SUPERUSER LOGIN PASSWORD 'npgsql_tests'"
+        pgsql/bin/psql -U postgres -c "CREATE DATABASE npgsql_tests OWNER npgsql_tests"
+        pgsql/bin/psql -U postgres -c "CREATE EXTENSION citext" npgsql_tests
+        pgsql/bin/psql -U postgres -c "CREATE EXTENSION hstore" npgsql_tests
+        pgsql/bin/psql -U postgres -c "CREATE EXTENSION ltree" npgsql_tests
+        pgsql/bin/psql -U postgres -c "CREATE EXTENSION postgis" npgsql_tests
+
+        # user 'npgsql_tests_scram' must be created with password encrypted as scram-sha-256 (which only applies after restart)
+        sed -i "s|#password_encryption = md5|password_encryption = scram-sha-256|" pgsql/PGDATA/postgresql.conf
+
+        pgsql/bin/pg_ctl -D pgsql/PGDATA -l logfile -o '-c max_prepared_transactions=10 -c ssl=true -c ssl_cert_file=../server.crt -c ssl_key_file=../server.key' restart
+
+        pgsql/bin/psql -U postgres -c "CREATE ROLE npgsql_tests_scram SUPERUSER LOGIN PASSWORD 'npgsql_tests_scram'"
+
+        # Disable trust authentication except for unix domain sockets, requiring MD5
+        # passwords - some tests must fail if a password isn't provided.
+        if [ ${{ variables.pg_major }} -ge 13 ]; then
+              echo "local all all trust" > pgsql/PGDATA/pg_hba.conf
+              echo "host all npgsql_tests_scram all scram-sha-256" >> pgsql/PGDATA/pg_hba.conf
+        else
+              echo "host all npgsql_tests_scram all scram-sha-256" > pgsql/PGDATA/pg_hba.conf
+        fi
+        echo "host all all all md5" >> pgsql/PGDATA/pg_hba.conf
+        echo "host replication all all md5" >> pgsql/PGDATA/pg_hba.conf
+      displayName: Start PostgreSQL ${{ variables.pg_major }} (Windows)
+      condition: startsWith(variables['imageName'], 'windows')
+
+    - script: |
+        dd-trace --dd-env CI --dd-service npgsql.tests --dd-version v4.1.9 dotnet "test npgsql/test/Npgsql.Tests --logger trx;LogFileName=npgsql.tests.$(tagName).trx --results-directory npgsql/test/results"
+      env:
+        DOTNET_SKIP_FIRST_TIME_EXPERIENCE: true
+      displayName: Test
+
+    - script: |
+        dd-trace --dd-env CI --dd-service npgsql.plugintests --dd-version v4.1.9 dotnet "test npgsql/test/Npgsql.PluginTests --logger trx;LogFileName=npgsql.plugintests.$(os).$(tagName).trx --results-directory npgsql/test/results"
```

Non blocking, just a personal preference...

I think you can avoid the `"` by using a `--` before the final command to wrap.
Also I prefer to use `=` to make all option values more clear.

```suggestion
        dd-trace --dd-env=CI --dd-service=npgsql.plugintests --dd-version=v4.1.9 -- dotnet test npgsql/test/Npgsql.PluginTests --logger trx;LogFileName=npgsql.plugintests.$(os).$(tagName).trx --results-directory npgsql/test/results
```

---

Thanks for the suggestion, I applied the `option=value`.
For the `"` style, I tried to remove them but the command line was interpreting `-- results-directory` as a separate command after running the `dd-trace` command, so I put them back